website/types.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright 2018 Ryan Dahl <ry@tinyclouds.org>
+// All rights reserved. MIT License.
+
+import * as ts from "typescript";
+
+export interface TSKit {
+  sourceFile: ts.SourceFile;
+  checker: ts.TypeChecker;
+  visited: Map<ts.Node, true>;
+  privateNames: One2ManyMap<string, any>;
+  typeParameters: string[];
+}
+
+export type Visitor = (this: TSKit, docEntries: any[], node: ts.Node) => void;
```

Comment:
"this" is not a good argument name. (The linter should forbid this. Extra points if you can fix the linter.)

Reply:
this is not a argument name, it just defines the type of `this`